.github/ISSUE_TEMPLATE/release-template.md

```diff
@@ -8,21 +8,49 @@ labels: 'Type: Maintenance'
 
 :warning: DO NOT MERGE (YET) :warning:
 
-[Remaining work for this Milestone](https://github.com/Parsely/wp-parsely/milestone/14)
+[Remaining work for this Milestone](https://github.com/Parsely/wp-parsely/milestone/xx)
 
 PR for tracking changes for the X.Y.Z release. Target release date: DOW DD MMMM YYYY.
 
-- [ ] Merge any outstanding PRs due for this release to `develop`.
+## Before releasing
+
 - [ ] Notify stakeholders of an upcoming release.
-- [ ] Add [PR](https://github.com/Parsely/wp-parsely/pull/XXX) against `develop` which:
-  - Increases version numbers in the README, plugin bootstrap file `wp-parsely.php`, and `package.json`.
-  - Adds or updates the changelog for the release.
-- [ ] Merge this PR.
-- [ ] Create a branch from `trunk` named `release-x.y.z`. Merge `develop` into that branch. Open a PR from `release-x.y.z` to `trunk` named _Release x.y.z_.
-- [ ] Add signed release tag against `trunk` using GitHub's UI.
+- [ ] Merge any outstanding PRs due for this release to `develop`.
+- [ ] Verify that all PRs have a `Changelog` tag.
+
+## Release proccess
+
+**Create the release PR**
+- [ ] Create a PR that merges `develop` into `trunk`, named _Release x.y.z_.
```

Comment:
Food for thought: could we add a template for the description of this kind of PRs?

Reply:
I think we could, given that texts are pretty standard. Do you want me to create an issue for it?

Reply:
Maybe it would also be possible to integrate this into the workflow.

Reply:
We can create an issue and integrate it in the workflow in the future.